src/hikari_cp/core.clj

```diff
@@ -181,11 +181,11 @@
       (.setMinimumIdle         minimum-idle)
       (.setMaximumPoolSize     maximum-pool-size))
     (when datasource (.setDataSource config datasource))
-    (when datasource-classname (.setDataSourceClassName config datasource-classname))
     (if adapter
```

Comment:
I would think splitting this `if` into two separate `when` would be better. Is there any reason why this was written like this?

Reply:
No particular reason I suppose

Reply:
Currently If you dont provide `adapter` or `jdbc-url`, it will fail with an NPE.
Shall I update it in this PR? or you would want it in a separate PR?

Reply:
Separate one, please :) Thanks!